src/vcpkg/base/files.cpp

```diff
@@ -857,36 +939,124 @@ namespace vcpkg
             }
         }
 
-        virtual std::vector<path> get_files_recursive(const path& dir) const override
+        template<class Iter>
+        static std::vector<path> get_files_impl(const path& dir, std::error_code& ec)
         {
             std::vector<path> ret;
-
-            std::error_code ec;
-            stdfs::recursive_directory_iterator b(dir, ec), e{};
-            if (ec) return ret;
-            for (; b != e; ++b)
+            Iter b(dir, ec), e{};
+            while (b != e)
             {
+                if (ec)
+                {
+                    ret.clear();
+                    break;
+                }
+
                 ret.push_back(b->path());
+                b.increment(ec);
```

Comment:
Bug: we lost the ec by checking b != e before checking ec again? Repeated 2x below.